pulser-core/pulser/waveforms.py

```diff
@@ -69,14 +82,9 @@ def __init__(self, duration: Union[int, Parametrized]):
         Args:
             duration: The waveforms duration (in ns).
         """
-        duration = cast(int, duration)
-        try:
-            _duration = int(duration)
-        except (TypeError, ValueError):
-            raise TypeError(
-                "duration needs to be castable to an int but "
-                f"type {type(duration)} was provided."
-            )
+        assert not isinstance(duration, Parametrized)
```

Comment:
Can't the duration be Parametrized ? I am puzzled with the type of `duration` that is `Union[int, Parametrized]`...

Reply:
Okay, so line 66 of this file you keep the "Parametrized" in `def __init__(self, duration: Union[int, Parametrized]):` only for the description ?

Reply:
Resolved.